### PR TITLE
Windows & Ubuntu CLANG CI support

### PR DESCRIPTION
**Pull request by dc-dc-dc**
Main changes were converting uses of '/tmp' to using a helper function that creates os specific paths with temp directory. Only had to skip one test(test_conv_transpose2d) because it was running out of memory in the runner.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -95,6 +95,27 @@ jobs:
...
+    - name: Checkout Code
+      uses: actions/checkout@v3
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.8
+    - name: Install Dependencies
+      run: pip install -e '.[testing]' --extra-index-url https://download.pytorch.org/whl/cpu
+    - name: Set env
+      run: printf "CI=1\nCLANG=1\nENABLE_METHOD_CACHE=1" >> $GITHUB_ENV
```
Comment (geohot) [inline .github/workflows/test.yml:116]: ENABLE_METHOD_CACHE is the default, you don't need to explicitly turn it on (should be removed from LLVM).

I think `CI=1` is automatic too in CI

**Comment by geohot** [comment]
Nice! Will merge as is, the extra flags don't really matter.

File: .github/workflows/test.yml (hunk 1 above)
Comment (dc-dc-dc) [reply .github/workflows/test.yml:116]: Just checked github docs and `CI=true` by default so can remove it as well as ENABLE_METHOD_CACHE

Comment (geohot) [reply .github/workflows/test.yml:116]: Yea, I'd merge a small PR cleaning it up, but no big deal.

**Pull request merged**